src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanFL401ConditionalTransformerService.java

```diff
@@ -78,4 +187,91 @@ private String transformReasonForOrderWithoutGivingNotice(Map<String, String> in
+                return relationship.getValue();
+            }
+        }
+        return null;
+    }
+
+    private TreeMap<String, String> buildRelationshipDetailMap() {
+        TreeMap<String, String> relationshipFieldMap = new TreeMap<>();
+
+        relationshipFieldMap.put(
+                RESPONDENT_MARRIED_CIVIL_RELATIONSHIP_FIELD, "Married or in a civil partnership");
+        relationshipFieldMap.put(
+                RESPONDENT_FORMERLY_MARRIED_CIVIL_RELATIONSHIP_FIELD,
+                "Formerly married or in a civil partnership");
+        relationshipFieldMap.put(
+                RESPONDENT_ENGAGED_PROPOSED_CIVIL_RELATIONSHIP_FIELD,
+                "Engaged or proposed civil partnership");
+        relationshipFieldMap.put(
+                RESPONDENT_FORMERLY_ENGAGED_PROPOSED_CIVIL_RELATIONSHIP_FIELD,
+                "Formerly engaged or proposed civil partnership");
+        relationshipFieldMap.put(
+                RESPONDENT_LIVE_TOGETHER_AS_COUPLE_FIELD, "Live together as a couple");
+        relationshipFieldMap.put(
+                RESPONDENT_FORMERLY_LIVE_TOGETHER_AS_COUPLE_FIELD,
+                "Formerly lived together as a couple");
+        relationshipFieldMap.put(
+                RESPONDENT_BOY_GIRL_FRIEND_PARTNER_LIVEWITHME_FIELD,
+                "Boyfriend, girlfriend or partner who does not live with me");
+        relationshipFieldMap.put(
+                RESPONDENT_FORMER_BOY_GIRL_FRIEND_PARTNER_LIVEWITHME_FIELD,
+                "Former boyfriend, girlfriend or partner who did not live with me");
+        relationshipFieldMap.put(
+                APPLICANT_RESPONDENT_OTHER_RELATIONSHIP_FIELD, "None of the above");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_FATHER_RELATIONSHIP_FIELD, "Father");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_MOTHER_RELATIONSHIP_FIELD, "Mother");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_SON_RELATIONSHIP_FIELD, "Son");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_DAUGHTER_RELATIONSHIP_FIELD, "Daughter");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_BROTHER_RELATIONSHIP_FIELD, "Brother");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_SISTER_RELATIONSHIP_FIELD, "Sister");
+        relationshipFieldMap.put(
+                APPLICANT_RESPONDENT_GRANDFATHER_RELATIONSHIP_FIELD, "Grandfather");
+        relationshipFieldMap.put(
+                APPLICANT_RESPONDENT_GRANDMOTHER_RELATIONSHIP_FIELD, "Grandmother");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_UNCLE_RELATIONSHIP_FIELD, "Uncle");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_AUNT_RELATIONSHIP_FIELD, "Aunt");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_NEPHEW_RELATIONSHIP_FIELD, "Nephew");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_NIECE_RELATIONSHIP_FIELD, "Niece");
+        relationshipFieldMap.put(APPLICANT_RESPONDENT_COUSIN_RELATIONSHIP_FIELD, "Cousin");
+        relationshipFieldMap.put(
```

Comment:
Move to enum class and get value based xl tag value is yes

Reply:
I'm using the map instead of enum to avoid too many conditional statement.

Reply:
Done. Adapted enum to the map.